packages/mui-system/src/createStyled.test.js

```diff
@@ -48,6 +48,22 @@ describe('createStyled', () => {
 
       expect(SomeMuiComponent).to.have.property('displayName', 'Styled(Component)');
     });
+
+    it('does not forward invalid props to DOM if no `slot` specified', () => {
+      // This scenario is usually used by library consumers
+      const styled = createStyled({});
+      const Anchor = styled('a')({});
+
+      const { container } = render(
+        <Anchor href="/" color="red" shouldBeRemoved data-foo="bar">
```

Comment:
```suggestion
        <Anchor href="/" color="red" shouldBeRemoved data-foo="bar">
```

The `href` is messing up the browser tests, maybe use a `button`?